pkg/endpoint/bpf.go

```diff
@@ -1406,14 +1427,38 @@ func (e *Endpoint) startSyncPolicyMapController() {
 					return controller.NewExitReason("Endpoint disappeared")
 				}
 				defer e.unlock()
-				return e.syncPolicyMapWithDump()
+				defer e.swapPolicyMapApplied()
+
+				err := e.syncPolicyMapWithDump(ctx)
```

Comment:
Any thoughts on the other `syncPolicyMap` methods and how they're used? Can we fold those into this one as well?